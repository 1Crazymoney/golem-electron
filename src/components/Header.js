import React, { Component } from 'react'
import { findDOMNode } from 'react-dom'
import { browserHistory, Link } from 'react-router'
/**
 * @see http://react-component.github.io/tooltip/
 */
import ReactTooltip from 'rc-tooltip'
const {BrowserWindow} = window.require('electron').remote;
import { setConfig, getConfig } from './../utils/configStorage'
/**
 * { Class for header component with navigation. }
 *
 * @class      Header (name)
 */
export default class Header extends Component {

    constructor(props) {
        super(props)
    }

    componentDidMount() {
        const index = location.pathname === "/" ? 1 : 2 // <-- HARDCODED
        let navItems = document.getElementsByClassName('nav__item')
        navItems[index].classList.add('active')
    }

    /**
     * [_navigateTo active class handling for navigation items]
     * @param  {String}     to      [Route fo the page]
     * @param  {Object}     elm     [Element in target]
     */
    _navigateTo(to, isNav = true, elm) {
        let navItems = document.getElementsByClassName('nav__item')
        let menuItems = document.getElementsByClassName('menu__item')
        if (isNav) {
            for (var i = 0; i < navItems.length; i++) {
                navItems[i].classList.remove('active')
            }
            for (var i = 0; i < menuItems.length; i++) {
                menuItems[i].classList.remove('active')
            }
            elm.currentTarget.classList.add('active')
        } else {
            for (var i = 0; i < navItems.length; i++) {
                navItems[i].classList.remove('active')
            }
            for (var i = 0; i < menuItems.length; i++) {
                menuItems[i].classList.remove('active')
            }
            elm.currentTarget.classList.add('active')
        }

        browserHistory.push(to);
    }


    /**
     * [_onClose,_onMinimize,_onMaximize Native Window Button handlers]
     */
    _onClose() {
        let win = BrowserWindow.getFocusedWindow();
        win.close()
    }

    _onMinimize() {
        let win = BrowserWindow.getFocusedWindow();
        win.minimize()
    }

    _onMaximize() {
        let win = BrowserWindow.getFocusedWindow();
        win.isMaximized() ? win.unmaximize() : win.maximize()
    }

    render() {
        return (
            <header className="header">
             <div className="top-titlebar">
                <div>
                    <span>Golem</span>
                </div>
                <div className="os__menu">
                    <span className="icon-minimize"/>
                    <span className="icon-close"/>
                </div>
             </div>
              <nav className="nav">
                <ul className="nav__list">
                    <li className="nav__item traffic-light">
                        <div className="close" onClick={::this._onClose}></div>
                        <div className="minimize" onClick={::this._onMinimize}></div>
                        <div className="maximize" onClick={::this._onMaximize} disabled={true}></div>
                    </li>
                    <li className="nav__item" onClick={this._navigateTo.bind(this, '/', true)}>Network</li>
                    <li className="nav__item" onClick={this._navigateTo.bind(this, '/tasks', true)}>Tasks</li>
                    <span className="selector"></span>
                </ul>
                <ul className="menu">
                    <ReactTooltip placement="bottom" trigger={['hover']} overlay={<p>New Task</p>}>
                        <li className="menu__item"><span className="icon-add"/></li>
                    </ReactTooltip>
                    <ReactTooltip placement="bottom" trigger={['hover']} overlay={<p>Docs</p>}>
                        <li className="menu__item"><span className="icon-doc"/></li>
                    </ReactTooltip>
                    <ReactTooltip placement="bottomRight" trigger={['hover']} overlay={<p>Settings</p>}>
                        <li className="menu__item" onClick={this._navigateTo.bind(this, '/settings', false)}><span className="icon-settings"/></li>
                    </ReactTooltip>
                </ul>
            </nav>
            </header>
        );
    }
}