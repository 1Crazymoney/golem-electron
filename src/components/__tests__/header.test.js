jest.unmock('../Header')

import React from 'react'
import ReactDOM from 'react-dom'
import TestUtils from 'react-addons-test-utils'
import Header from '../Header'


it('should render a header component', () => {
    const wrapper = shallow(
        <Header/>
    )
    expect(wrapper).toMatchSnapshot()
})