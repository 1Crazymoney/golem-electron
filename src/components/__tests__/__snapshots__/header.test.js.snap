// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a header component 1`] = `
<header
  className="header "
>
  <nav
    className="nav nav-testnet"
    role="menubar"
  >
    <div
      className="draggable draggable--other"
      style={
        Object {
          "WebkitAppRegion": "drag",
        }
      }
    />
    <ConditionalRender
      showIf={false}
    >
      <div
        className="nav__list"
      >
        <img
          className="logo__header"
          src=""
        />
      </div>
      <ul
        className="menu"
        role="menu"
      >
        <ForwardRef(TippyWrapper)
          content={
            <div
              className="menu__upload"
            >
              <div
                className="menu-item__upload"
                onClick={[Function]}
              >
                <span
                  aria-label="Add file for task"
                  className="icon-file"
                  role="menuitem"
                  tabIndex="0"
                  title="Add File"
                />
                <span>
                  Add file
                </span>
              </div>
              <div
                className="menu-item__upload"
                onClick={[Function]}
              >
                <span
                  aria-label="New folder for task"
                  className="icon-folder"
                  role="menuitem"
                  tabIndex="0"
                  title="Add Folder"
                />
                <span>
                  Add folder
                </span>
              </div>
            </div>
          }
          hideOnClick={true}
          interactive={true}
          isEnabled={false}
          placement="bottom"
          theme="light"
          trigger="click"
        >
          <ForwardRef(TippyWrapper)
            content={
              <p>
                Golem is not started yet.
              </p>
            }
            placement="bottom"
            trigger="mouseenter"
          >
            <li
              className="menu__item upload-menu"
            >
              <span
                aria-label="New Task"
                className="icon-add"
                role="menuitem"
                tabIndex="0"
                title="Upload file for the task"
              />
            </li>
          </ForwardRef(TippyWrapper)>
        </ForwardRef(TippyWrapper)>
        <ForwardRef(TippyWrapper)
          content={
            <p>
              Docs
            </p>
          }
          placement="bottom"
          trigger="mouseenter"
        >
          <li
            className="menu__item"
          >
            <a
              href="https://docs.golem.network/"
            >
              <span
                aria-label="Documentation"
                className="icon-doc"
                role="menuitem"
                tabIndex="0"
              />
            </a>
          </li>
        </ForwardRef(TippyWrapper)>
        <ForwardRef(TippyWrapper)
          content={
            <p>
              Settings
            </p>
          }
          placement="bottom"
          trigger="mouseenter"
        >
          <li
            aria-label="Settings"
            className="menu__item"
            data-path="/settings"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
          >
            <span
              className="icon-settings"
            />
          </li>
        </ForwardRef(TippyWrapper)>
      </ul>
    </ConditionalRender>
    <ConditionalRender
      showIf={false}
    >
      <div
        className="header__frame"
      >
        <div
          className="title"
        >
          <span />
        </div>
        <div
          className="info"
        >
          <span
            className="time"
          />
          <span
            className="amount__frame"
          >
             
             Frame
          </span>
        </div>
        <div
          className="menu"
          role="menu"
        >
          <span
            aria-label="Completed Frames"
            className="menu__item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
          >
            Complete
          </span>
          <span
            aria-label="All Frames"
            className="menu__item "
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
          >
            All
          </span>
        </div>
      </div>
    </ConditionalRender>
  </nav>
  <ConditionalRender
    showIf={false}
  >
    <nav
      className="nav"
    >
      <ul
        className="nav__list"
        role="menu"
      >
        <li
          aria-label="Network"
          className="nav__item"
          data-path="/"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          Network
        </li>
        <li
          aria-label="Tasks"
          className="nav__item"
          data-path="/tasks"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          Tasks
        </li>
        <span
          className="selector"
        />
      </ul>
    </nav>
  </ConditionalRender>
</header>
`;
